packages/nextjs/src/config/loaders/rollup.ts

```diff
@@ -0,0 +1,110 @@
+  //     root. Unclear why it's not.)
+  //   - As a result of the miscalculation, our proxy module will include `export { helperFunc } from '../utils/helper'`
+  //     rather than the expected `export { helperFunc } from '../../utils/helper'`, thereby causing a build error in
+  //     nextjs..
+  //
+  // It's not 100% clear why, but telling it not to do the conversion back from absolute to relative (by setting
+  // `makeAbsoluteExternalsRelative` to `false`) seems to also prevent it from going from relative to absolute in the
+  // first place, with the result that the path remains untouched (which is what we want.)
+  makeAbsoluteExternalsRelative: false,
+});
+
+const rollupOutputOptions: RollupOutputOptions = {
+  format: 'esm' as const,
+
+  // Don't create a bundle - we just want the transformed entrypoint file
+  preserveModules: true,
+};
+
+/**
+ * Use Rollup to process the proxy module file (located at `tempProxyFilePath`) in order to split its `export * from
+ * '<wrapped file>'` call into individual exports (which nextjs seems to need).
+ *
+ * @param tempProxyFilePath The path to the temporary file containing the proxy module code
+ * @param resourcePath The path to the file being wrapped
+ * @returns The processed proxy module code, or undefined if an error occurs
+ */
+export async function rollupize(tempProxyFilePath: string, resourcePath: string): Promise<string | undefined> {
+  let finalBundle;
+
+  try {
+    const intermediateBundle = await rollup(getRollupInputOptions(tempProxyFilePath, resourcePath));
+    finalBundle = await intermediateBundle.generate(rollupOutputOptions);
+  } catch (err) {
+    __DEBUG_BUILD__ &&
+      logger.warn(
+        `Could not wrap ${resourcePath}. An error occurred while processing the proxy module template:\n${err}`,
+      );
+    return undefined;
+  }
+
+  // The module at index 0 is always the entrypoint, which in this case is the proxy module.
+  let { code } = finalBundle.output[0];
+
+  // Rollup does a few things to the code we *don't* want. Undo those changes before returning the code.
+
+  // If the wrapped file only has a default export, our `export * from '...'` won't have anything in it (`export *`
+  // never includes the default export, which must be handled separately). In order to preserve whatever side effects
+  // running the file which is the source of the exports might have, though, Rollup will still emit an `import '...'
+  // statement. We're importing the same file later on in the proxy template, so we don't need this.
+  const sideEffectsImportRegex = new RegExp("import '.*';", 'g');
+  code = code.replace(sideEffectsImportRegex, '');
+
+  // Nextjs uses square brackets surrounding a path segment to denote a parameter in the route, but rollup turns those
+  // square brackets into underscores. Further, Rollup adds file extensions to bare-path-type import and export sources.
```

Comment:
> rollup turns those square brackets into underscores

wat

![Jackie-Chan-WTF](https://user-images.githubusercontent.com/8118419/185380653-fd57f25b-208c-4837-ae19-6e57b82b0ccc.jpeg)

Reply:
Yeah, I know. I don't understand how that could possibly work, for anyone. I feel like it's gotta be a bug, which I should probably report...